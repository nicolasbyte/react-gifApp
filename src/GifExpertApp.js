import React, { useState } from 'react';
import { AddCategorie } from './components/AddCategorie';
import { GifGrid } from './components/GifGrid';



const GifExpertApp = ( { categoriesDefault = [] } ) => {
    
        const [ categories , setCategories ] = useState( categoriesDefault );


        // const handlAdd = () => {

        //     const add = 'Naruto';

        //     setCategories( [ ...categories , add ] );
        // }

        return (
            <>

                <h1>GifExpertApp</h1>
                <hr />

                <AddCategorie setCategories = { setCategories } />

                {/* <button onClick= { handlAdd }>
                    Agregar
                </button> */}

                <ol>
                    {
                        categories.map( c => (
                            <GifGrid
                                    key      = { c } 
                                    category = { c } 
                                    
                                    />
                        ))
                    }
                </ol>
            </>
        );
};


export default GifExpertApp;