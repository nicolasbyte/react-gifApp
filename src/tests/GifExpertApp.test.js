import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('este es el test de GifExpertApp', () => {
    
    test('este es el snapshot contra el GifExpertApp', () => {
        
            const wrapper = shallow( <GifExpertApp />);

            expect( wrapper ).toMatchSnapshot();
    });


    test('debe de mostrar una lista de categorias', () => {
       
        const categories = [ 'One Punch' , 'Naruto' ];

        const wrapper = shallow( < GifExpertApp  categoriesDefault = { categories } /> );

        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
    
});
