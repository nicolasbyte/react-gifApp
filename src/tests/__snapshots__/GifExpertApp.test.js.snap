// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`este es el test de GifExpertApp este es el snapshot contra el GifExpertApp 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategorie
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
